src/main/native/compression/IntelDeflater.cc

```diff
@@ -323,7 +323,8 @@ JNIEXPORT jint JNICALL Java_com_intel_gkl_compression_IntelDeflater_deflateNativ
     }
 
     // return number of bytes written to output buffer
-    long bytes_out = outputBufferLength - lz_stream->avail_out;
+    // ISAL is capable of handling 64 bit buffer sizes but we're truncating to stay consistent with the Java interface.
```

Comment:
Please check 64 is the right value in the comment.

Reply:
I removed this comment completely as no type conversions are actually being performed here.